### ISSUE TITLE
Dataset QC manual verification and selection for training final MONAI segmentation model

### ISSUE THREAD

**Issue by rohanbanerjee** [sct-pipeline/fmri-segmentation#21]
I am opening this issue to have the discussion around selection of the datasets to be included for training based on the quality of the ground truths. The list of all the datasets can be found [here].(https://docs.google.com/spreadsheets/d/1xZMuR5OLRRIRWyIJicIqr6znDAaaJUqaOhgLX8qWgJI/edit?usp=sharing)

### Current status:
- Shared SCT QC for all the datasets with @MerveKaptan 
- @MerveKaptan is going through the QCs

**Comment by MerveKaptan** [comment]
Dear all,

Based on the QCs that @rohanbanerjee shared with me, I suggest these datasets to be included in the model:

- Geneva
- Leipzig rest
- NWMotor
- NWMotorWeber
- NWTactile
- NWThermal
- Zurich Cervical

I am ambiguous about Stanford rest and Stanford rest Martucci. Stanford Rest is due to data quality (although masks are fine). Please see an example below:
_without mask_
![image](https://github.com/sct-pipeline/fmri-segmentation/assets/46893456/d99ff096-e0dc-475e-ab1f-a3868cfc4bdd)
_with mask_
![image](https://github.com/sct-pipeline/fmri-segmentation/assets/46893456/dfbd4d33-5a1d-41c5-8ab7-14d47a423d9e)

I think Stanford rest Martucci's masks are too big and extends beyond the cord. Please see an example below:
![image](https://github.com/sct-pipeline/fmri-segmentation/assets/46893456/bbbe8813-0fe5-4a95-960b-d048692131ec)
![image](https://github.com/sct-pipeline/fmri-segmentation/assets/46893456/d1a78c2b-63e6-4271-8436-e4359d7e4f60)

**Comment by jcohenadad** [comment]
> with mask

looks ok-- slightly undersegmented to my taste-- last slice mask is missing

> I think Stanford rest Martucci's masks are too big and extends beyond the cord. Please see an example below:

waaaay oversegmented-- 

next time pls show the GIF-- easier to assess

**Comment by rohanbanerjee** [comment]
Thank you for the verification. So as it stands, we have to just remove the `kcl_rest` set for the the `multi-site` dataset we were already using. 

Another thing to note here is that, there are cases in which we could you parts of the segmentation in the training set. The following cases are examples of what I mean:

1. In the whole image, there is a missing ground truth slice/only a few slices which are not good quality.
2. There are places where the SC is not clearly visible and the ground truth is still drawn.

In case 1, these slices can be rectified and used for training.
In case 2, the ground truth from these slices can either be removed or a new ground truth can be draw for these specific images.

Pro of what I mentioned -- we have more variability and in the training set.
Con -- This is can introduce biases.

**Comment by MerveKaptan** [comment]
> looks ok-- slightly undersegmented to my taste-- last slice mask is missing
> 
> >

The first and last slices are not used in the further analyses - this is why they are not segmented!

**Comment by jcohenadad** [comment]
> The first and last slices are not used in the further analyses - this is why they are not segmented!

Right, but the DL model doesn't know this, so it contributes to the under-performance of the model. @rohanbanerjee pls make sure to address this
